flaml/searcher/blendsearch.py

```diff
@@ -267,25 +289,18 @@ def on_trial_complete(self, trial_id: str, result: Optional[Dict] = None,
             else:  # add to result cache
                 self._result[self._ls.config_signature(config)] = result
                 # update target metric if improved
-                objective = result[
-                    self._metric + self.lagrange] if self._metric_constraints \
-                    else result[self._metric]
+                objective = result[self._ls.metric]
                 if (objective - self._metric_target) * self._ls.metric_op < 0:
                     self._metric_target = objective
                 if not thread_id and metric_constraint_satisfied \

```

Comment:
can we change 'not thread_id' to 'thread_id==0'?

Reply:
We can't. thread_id could be None.